tutorials/notebook.md

```diff
@@ -0,0 +1,767 @@
+already created a wallet and saved some notes.
+
+
+In file `notebook/index.js`:
+```javascript
+function showNotes(creds) {
+  if (!idFileExists()) {
+    console.log("Error: you must register before you can print notes.");
+
+    return;
+  }
+
+  let identifiers = readIdentifierFile();
+  let client = connect(creds);
+  let path = CHAIN_TREE_NOTE_PATH;
+
+  client.resolve(identifiers.chainId, CHAIN_TREE_NOTE_PATH)
+    .then(function(resp) {
+      let notes = resp.data;
+
+      if (notes instanceof Array) {
+        console.log('----Notes----');
+        notes.forEach(function(note) {
+          console.log(note);
+        });
+      } else {
+        console.log('----No Notes-----');
+      }
+    }, function(err) {
+      console.log('Error fetching notes: '  + err);
+    });
+}
+```
+
+## Adding a Command Line Interface
+Now that we have all the background functions we need to manage notes with our
+application, let's add a command line interface to tie everything together.
+We'll use the [Yargs](https://github.com/yargs/yargs "Yargs") library for the
+CLI, so let's add it as a dependency to our `package.json` and require it at the
+top of our `index.js` file.
+
+
+In file `notebook/package.json`:
+```javascript
+{
+    "name": "notebook",
+    ...
+    "dependencies": {
+        "tupelo-client": "^0.0.2-alpha1",
+        "yargs": "^12.0.2"
+    }
+}
+```
+
+In file `notebook/index.js`:
+```javascript
+const tupelo = require('tupelo-client');
+const fs = require('fs');
+const yargs = require('yargs');
+...
+```
+
+Next we'll follow the `yargs` documentation to define 3 commands: `register`,
+`add-note`, and `print-notes`.
+
+In file `notebook/index.js`:
+```javascript
+yargs.command('register [name] [passphrase]', 'Register a new notebook chain tree', (yargs) => {
+  yargs.positional('name', {
+    describe: 'Name of the wallet to save the notes chain tree.'
+  }).positional('passphrase', {
+    describe: 'Wallet passphrase.'
+  });
+}, (argv) => {
+  let creds = {
+    walletName: argv.name,
+    passPhrase: argv.passphrase
+  };
+
+  createNotebook(creds);
+
+}).command('add-note [name] [passphrase]', 'Save a note', (yargs) => {
+  yargs.positional('name', {
+    describe: 'Name of the wallet where the notes chain tree is saved.'
+  }).positional('passphrase', {
+    describe: 'Wallet passphrase.'
+  }).describe('n', 'Save a note')
+    .alias('n', 'note')
+    .demand('n');
+}, (argv) => {
+  let creds = {
+    walletName: argv.name,
+    passPhrase: argv.passphrase
+  };
+
+  addNote(creds, argv.n);
+
+}).command('print-notes [name] [passphrase]', 'Print saved notes', (yargs) => {
+  yargs.positional('name', {
+    describe: 'Name of the wallet where the notes chain tree is saved.'
+  }).positional('passphrase', {
+    describe: 'Wallet passphrase.'
+  });
+}, (argv) => {
+  let creds = {
+    walletName: argv.name,
+    passPhrase: argv.passphrase
+  };
+
+  showNotes(creds);
+}).argv;
+```
+
+## Finishing Up
```

Comment:
I think I would put the "final file" in a separate page with just a link here.
I would also add a few example command lines "node index.js register notebook_name notebook_pw" or whatever.

Reply:
+1 to final invocation examples. A create notebook, add note or two, then show all notes workflow would make sense to me.